catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.async.data.impl;
+
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.CONTENT_SCHEME;
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.DEFAULT_MIME_TYPE;
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.DEFAULT_NAME;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.function.Supplier;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.resource.Resource;
+
+public class LazyProcessResourceImpl implements ProcessResource {
```

Comment:
The added javadoc.
The difference is that LazyProcessResourceImpl is used when the resource hadn't already been loaded and may not be needed.
The ProcessResourceImpl assumes the resource had already been loaded and the resource is used to create the ProcessResourceImpl.
I think we should keep both because of the difference. I've heard work was being done in other areas where the resource has been loaded. So the ProcessResourceImpl would be used in that case.